keras_nlp/benchmarks/text_generation.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Benchmark for text generation."""
+
+import time
+
+import tensorflow as tf
+from tensorflow import keras
+
+import keras_nlp
+from keras_nlp.benchmarks.text_generation_config import COMMON_ARGS
+from keras_nlp.benchmarks.text_generation_config import TEST_RUNS
+from keras_nlp.utils import beam_search
+from keras_nlp.utils import greedy_search
+from keras_nlp.utils import random_search
+from keras_nlp.utils import top_k_search
+from keras_nlp.utils import top_p_search
+
+SUPPORTED_TEXT_GEN_METHODS = {
+    "greedy_search": greedy_search,
+    "random_search": random_search,
+    "beam_search": beam_search,
+    "top_k_search": top_k_search,
+    "top_p_search": top_p_search,
+}
+EXECUTION_METHODS = ["graph", "xla"]
+
+
+def generate_random_ds(vocab_size, num_samples, batch_size, seed):
+    inputs = tf.random.uniform(
+        shape=(num_samples, 2),
+        minval=0,
+        maxval=vocab_size - 1,
+        dtype=tf.dtypes.int32,
+        seed=seed,
+    )
+
+    ds = tf.data.Dataset.from_tensor_slices(inputs)
+    ds = ds.batch(batch_size)
+    return ds
+
+
+def build_model(
+    vocab_size, max_length, embed_dim, num_layers, num_heads, ff_dim
+):
+    inputs = keras.layers.Input(shape=(None,), dtype=tf.int32)
+    # Embedding.
+    embedding_layer = keras_nlp.layers.TokenAndPositionEmbedding(
```

Comment:
could make this slightly more concise with the following style, as we don't need the intermediate layers

```
x = keras_nlp.layers.TransformerDecoder(
    num_heads=num_heads,
    intermediate_dim=ff_dim,
)(x)
```